lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -265,6 +276,63 @@ public void run() {
 		}
 	}
 
+	/**
+	 * Authentication token changes for the current connection are possible when
+	 * the client is in the {@link ConnectionState#connected}, {@link ConnectionState#connecting}
+	 * or {@link ConnectionState#disconnected } states.
+	 *
+	 * In the current protocol version we are not able to update auth params on the fly;
+	 * so disconnect, and the new auth params will be used for subsequent reconnection
+	 * * <p>
+	 * Spec: RTC8 (reauthorise)
+	 * </p>
+	 *
+	 */
+	public void onAuthUpdated() {
+		switch (state.state){
+			case connected:
+			case connecting:
+			case disconnected:
+				connection.once(ConnectionState.failed, new ConnectionStateListener() {
+					@Override
+					public void onConnectionStateChanged(ConnectionStateChange state) {
+						synchronized (ConnectionManager.this) {
+							ConnectionManager.this.notify();
+						}
+
+					}
+				});
+				disconnect();
+				synchronized (this){
+					try {
+						wait();
+					} catch (InterruptedException e) {
+						e.printStackTrace();
```

Comment:
If we think that this case is possible (is there anything in the code that would interrupt this wait?) then this case should include a log message via the `Logger` instead of an explicit `printStackTrace()`. If we don't think this is reachable (ie because nothing interrupts the `wait()`, or only does so in circumstances that mean this is not an error, then we should have an empty catch block with no log statement.

Afaik there may be an interrupt to unblock the waiting thread as part of the shutdown sequence of the library; or it might be by a `notify()` - I can't remember. Either way, I think you should find out and decide what to put here.